apps/src/dance/lottie/LottieDancerUtils.ts

```diff
@@ -464,65 +511,194 @@ export function loadCanvasAnimation(config: CanvasAnimConfig): AnimationItem {
 }
 
 /**
- * Resolves the generated dancer assets (head PNG + metadata JSON).
+ * Resolves the generated dancer assets (head PNG + metadata JSON) and optional body PNG and metadata JSON.
  * Source of truth order:
- *   1) URL params (?path=...&dancer=...) → use exactly that dancer
+ *   1) URL params (?path=...&dancer=...) use exactly that dancer and/or (?body=...) for body
  *   2) localStorage('dancer-ai-generate') with {adlibOption, choices[], variant}
  *   3) hardcoded fallbacks (DEFAULT_HEAD_URL / DEFAULT_METADATA_URL)
+ * The renderer will not replace the body unless a bodyUrl is returned here.
  * TODO: Use channel ID instead of local storage.
  */
-export function resolveDancerAssets(
-  opts: ResolveDancerAssetsOpts = {}
-): ResolvedDancerAssets {
-  const {sourceTag = 'adefaultpp'} = opts;
-
+export function resolveDancerAssets(opts: ResolveDancerAssetsOpts = {}): {
```

Comment:
This function was getting messy so I've attempted to clean it up. It will now start with a set of default URLs that can be selectively replaced with better values from `localStorage` or URL parameters. It has also been expanded to handle body URLs.